File: BandManagement/ClientApp/app/models/search.model.ts
export class SearchModel {
    PageIndex: number | null;
    RecordsPerPage: number | null;
    SortColumn: string | null;
    IsAscendingSort: boolean | null;
    Filters: SearchFilter[]
}

export class SearchFilter {
    Column: string | null;
    FilterValue: string | null;
    Type: number | null;
}

export class SearchResult<T> {
    Records: T[];
    TotalRecords: number;
}

File: Domain.Model/Infraestructure/SearchModel.cs
using System.Collections.Generic;

namespace Domain.Model.Infraestructure
{
    public class SearchModel
    {
        public SearchModel()
        {
            this.Filters = new List<FilterModel>();
        }

        public int? PageIndex { get; set; }

        public int? RecordsPerPage { get; set; }

        public string SortColumn { get; set; }

        public bool? IsAscendingSort { get; set; }

        public List<FilterModel> Filters {get;}
    }
}
